### ISSUE TITLE
balancer: existing LB policies must update picker synchronously

### ISSUE THREAD

**Issue by easwars** [grpc/grpc-go#5469]
https://github.com/grpc/grpc-go/issues/5211 talks about delaying picker updates in LB policies which have child policies, until the configs have been propagated to all the children. It also talks about auditing existing code to ensure that existing LB policy implementations update their pickers synchronously upon receipt of a config update.

We will use this issue to ensure that we have audited all existing LB policy implementations, and update docs to indicate the same.

- [ ] cds_experimental
- [ ] xds_cluster_impl_experimental
- [ ] xds_cluster_manager_experimental
- [ ] cluster_resolver_experimental
- [ ] outlier_detection_experimental
- [ ] priority_experimental
- [ ] ring_hash_experimental
- [ ] round_robin
- [ ] pick_first
- [ ] grpclb
- [ ] rls_experimental
- [ ] weighted_round_robin
- [ ] weighted_target_experimental
- [ ] update documentation to indicate that LB policies must update their pickers synchronously upon receipt of config udpate

**Comment by aranjans** [comment]
@easwars I see ringhash_experimental policy already has picker update in sync with the config update [here](https://github.com/grpc/grpc-go/blob/master/xds/internal/balancer/ringhash/ringhash.go#L302). Am I missing something here?

**Comment by easwars** [comment]
> @easwars I see ringhash_experimental policy already has picker update in sync with the config update [here](https://github.com/grpc/grpc-go/blob/master/xds/internal/balancer/ringhash/ringhash.go?rgh-link-date=2024-05-02T07%3A25%3A34Z#L302) and similarly i see the same for many other lb policies. Am I missing something here?

Yes, you are right about the ringhash LB policy. I updated the PR description to reflect this.

The same needs to be done for other policies in the list and once confirmed, the PR description needs to be updated accordingly. Thanks.

**Comment by aranjans** [comment]
@easwars What about `outlier_detection_experimental` [here](https://github.com/grpc/grpc-go/blob/master/xds/internal/balancer/outlierdetection/balancer.go#L330)?
